lib/backend/dynamo/dynamodbbk.go

```diff
@@ -655,9 +668,16 @@ func (b *Backend) getTableStatus(ctx context.Context, tableName string) (tableSt
 // following docs partial:
 // docs/pages/includes/dynamodb-iam-policy.mdx
 func (b *Backend) createTable(ctx context.Context, tableName string, rangeKey string) error {
-	pThroughput := dynamodb.ProvisionedThroughput{
-		ReadCapacityUnits:  aws.Int64(b.ReadCapacityUnits),
-		WriteCapacityUnits: aws.Int64(b.WriteCapacityUnits),
+	var pThroughput *dynamodb.ProvisionedThroughput
+	var billingMode *string
+	if *b.OnDemand {
+		billingMode = aws.String(dynamodb.BillingModePayPerRequest)
+	} else {
+		pThroughput = &dynamodb.ProvisionedThroughput{
+			ReadCapacityUnits:  aws.Int64(b.ReadCapacityUnits),
+			WriteCapacityUnits: aws.Int64(b.WriteCapacityUnits),
+		}
+		billingMode = aws.String(dynamodb.BillingModeProvisioned)
```

Comment:
I think this reads better
```suggestion
	billingMode := aws.String(dynamodb.BillingModeProvisioned)
	pThroughput := &dynamodb.ProvisionedThroughput{
		ReadCapacityUnits:  aws.Int64(b.ReadCapacityUnits),
		WriteCapacityUnits: aws.Int64(b.WriteCapacityUnits),
	}
	if *b.OnDemand {
		billingMode = aws.String(dynamodb.BillingModePayPerRequest)
		pThroughput = nil
	}
```